### PR TITLE
chore(linux): Improve Sentry environment setting

### PR DESCRIPTION
**Pull request by ermshiperete**
When running from source we set the environment to `local` when reporting to Sentry. This change also adds the package version and the tier as additional Sentry tags.

This is a follow-up of PR #4581.

### CODE REVIEW

File: linux/scripts/reconf.sh [hunk 1]
```diff
@@ -60,8 +60,10 @@ for proj in ${extra_projects}; do
         cd keyman-config
         make clean
         cd keyman_config
-        sed -e "s/_VERSION_/${VERSION}/g" -e "s/_VERSIONWITHTAG_/${VERSION_WITH_TAG}/g" -e "s/_MAJORVERSION_/${VERSION_MAJOR}/g" \
-            -e "s/_RELEASEVERSION_/${VERSION_RELEASE}/g" -e "s/_TIER_/${TIER}/g" version.py.in > version.py
+        sed -e "s/_VERSION_/${VERSION}/g" -e "s/_VERSIONWITHTAG_/${VERSION_WITH_TAG}/g" \
+            -e "s/_MAJORVERSION_/${VERSION_MAJOR}/g" \
+            -e "s/_RELEASEVERSION_/${VERSION_RELEASE}/g" -e "s/_TIER_/${TIER}/g" \
+            -e "s/_ENVIRONMENT_/${VERSION_ENVIRONMENT}/g" version.py.in > version.py
```
Comment (mcdurdin) [inline linux/scripts/reconf.sh:66]:
```suggestion
        sed \
            -e "s/_VERSION_/${VERSION}/g" \
            -e "s/_VERSIONWITHTAG_/${VERSION_WITH_TAG}/g" \
            -e "s/_MAJORVERSION_/${VERSION_MAJOR}/g" \
            -e "s/_RELEASEVERSION_/${VERSION_RELEASE}/g" \
            -e "s/_TIER_/${TIER}/g" \
            -e "s/_ENVIRONMENT_/${VERSION_ENVIRONMENT}/g" \
            version.py.in > version.py
```

**Review by mcdurdin — COMMENTED** [review]
I'm not seeing where `environment` is set to `local`?

**Comment by ermshiperete** [comment]
> I'm not seeing where `environment` is set to `local`?

We use the `VERSION_ENVIRONMENT` environment variable which gets set in `resources/build/build-utils.sh`.

**Comment by mcdurdin** [comment]
Good stuff. So do we support the 5 environment types we are using for the apps: `alpha`, `beta`, `stable`, `local`, `test`? `alpha`, `beta` and `stable` are for published builds; `test` is during test runners, and `local` for running from source or unofficial builds.

**Comment by ermshiperete** [comment]
yes

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.253-beta](https://keyman.com/downloads/releases/beta/14.0.253)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.19-alpha](https://keyman.com/downloads/releases/alpha/15.0.19)

**Pull request merged**